oletools/util/log_helper/log_helper.py

```diff
@@ -0,0 +1,194 @@
+"""
+log_helper.py
+
+General logging helpers
+
+.. codeauthor:: Intra2net AG <info@intra2net>
+"""
+
+# === LICENSE =================================================================
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#  * Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+# -----------------------------------------------------------------------------
+# CHANGELOG:
+# 2017-12-07 v0.01 CH: - first version
+# 2018-02-05 v0.02 SA: - fixed log level selection and reformatted code
+# 2018-02-06 v0.03 SA: - refactored code to deal with NullHandlers
+# 2018-02-07 v0.04 SA: - fixed control of handlers propagation
+# 2018-04-23 v0.05 SA: - refactored the whole logger to use an OOP approach
+
+# -----------------------------------------------------------------------------
+# TODO:
+
+
+from ._json_formatter import JsonFormatter
+from ._logger_adapter import OletoolsLoggerAdapter
+from . import _root_logger_wrapper
+import logging
+import sys
+
+
+LOG_LEVELS = {
+    'debug': logging.DEBUG,
+    'info': logging.INFO,
+    'warning': logging.WARNING,
+    'error': logging.ERROR,
+    'critical': logging.CRITICAL
+}
+
+DEFAULT_LOGGER_NAME = 'oletools'
+DEFAULT_MESSAGE_FORMAT = '%(levelname)-8s %(message)s'
+
+
+class LogHelper:
+    def __init__(self):
+        self._all_names = set()  # set so we do not have duplicates
+        self._use_json = False
+        self._is_enabled = False
+
+    def get_or_create_silent_logger(self, name=DEFAULT_LOGGER_NAME, level=logging.CRITICAL + 1):
```

Comment:
Keep the same behavior as before: loggers will by default have a level `logging.CRITICAL + 1` and a `NullHandler`. This function is called by modules when creating loggers[1](https://github.com/christian-intra2net/oletools/blob/7cb86312f0a9386926b56e9c8f5578090083edfe/oletools/msodde.py#L227)[2](https://github.com/christian-intra2net/oletools/blob/7cb86312f0a9386926b56e9c8f5578090083edfe/oletools/ooxml.py#L30).